migration/ram.c

```diff
@@ -842,7 +846,9 @@ static void migration_bitmap_sync(RAMState *rs)
     qemu_mutex_lock(&rs->bitmap_mutex);
     rcu_read_lock();
     RAMBLOCK_FOREACH(block) {
-        migration_bitmap_sync_range(rs, block, 0, block->used_length);
+        if (!migrate_bypass_shared_memory() || !qemu_ram_is_shared(block)) {
```

Comment:
I though `migrate_bypass_shared_memory` was enough

Reply:
Even when `migrate_bypass_shared_memory` is set, ram can be set private instead of being shared. The only case we can skip syncing is `migrate_bypass_shared_memory && ram_is_shared`.